lib/flight_scheduler/batch_script_runner.rb

```diff
@@ -73,8 +73,26 @@ def run
           Process::Sys.setuid(@job.username)
           Process.setsid
 
-          batchd = Batchd.new(@job, @script)
-          batchd.run
+          # We've inherited the running thread when we forked.  The runner
+          # thread contains a running `::Async::Reactor` which doesn't play
+          # nicely with `fork`.  We shut it down here and create a new thread
+          # so that we can safely start a new reactor.
+          #
+          # XXX This could all be avoided by execing into a new process here.
```

Comment:
I don't see any issue with closing the reactor like this. I'm not entirely sure what issue it could cause, as `batchd` and `stepd` would block the forked version, however it is probably best if it stops.

One thing is, we probably don't need to go through a graceful shutdown as the main process will continue to run. We can probably use `reactor.stop` and `reactor.stopped?` instead. This way it will stop running tasks ASAP.

I'm not sure what the second thread is for, but I assume their is a valid reason. Would something like this work:

```
reactor = ::Async::Task.curren.reactor
reactor.stop
Thread.new do
  sleep 0.1 until reactor.closed?
  ...
end.join
```

Or even better:

```
::Async::Task.current.reactor.stop
batchd = ....
```

Reply:
Unfortunately, neither of those work.  For some reason after a fork, running `Task.yield`, which gets called a lot by `async`, results in the main fibre getting upset and throwing an exception.  Creating a new thread works around that.  That's the part that I'm complaining about in the comment.

I don't imagine `stop(false)` would be noticeably quicker than `close`.  We can revisit if we discover performance issues.